src/lib.rs

```diff
@@ -1,744 +1,345 @@
+        }
     }
 
     pub fn get_reflection(
         &self,
         shader_stage: IRShaderStage,
         reflection: &mut IRShaderReflection,
     ) -> bool {
-        unsafe { (self.funcs.get_reflection)(self.me, shader_stage, reflection.me) }
+        unsafe {
+            self.funcs
+                .IRObjectGetReflection(self.me, shader_stage, reflection.me)
+        }
     }
 }
 
-#[derive(Debug, Clone)]
-struct IRMetalLibBinaryFn<'lib> {
-    create: libloading::Symbol<'lib, unsafe extern "C" fn() -> *mut IRMetalLibBinaryOpaque>,
-    destroy: libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRMetalLibBinaryOpaque)>,
-
-    get_bytecode: libloading::Symbol<
-        'lib,
-        unsafe extern "C" fn(*mut IRMetalLibBinaryOpaque, *mut u8) -> usize,
-    >,
-    get_bytecode_size:
-        libloading::Symbol<'lib, unsafe extern "C" fn(*mut IRMetalLibBinaryOpaque) -> usize>,
-
-    synthesize_stage_in_function: libloading::Symbol<'lib, unsafe extern "C" fn() -> ()>,
-}
-
-pub struct IRMetalLibBinary<'lib> {
-    me: *mut IRMetalLibBinaryOpaque,
-    funcs: IRMetalLibBinaryFn<'lib>,
+pub struct IRMetalLibBinary {
+    me: *mut bindings::IRMetalLibBinary,
+    funcs: Arc<bindings::metal_irconverter>,
 }
 
-impl<'lib> Drop for IRMetalLibBinary<'lib> {
+impl Drop for IRMetalLibBinary {
     fn drop(&mut self) {
-        unsafe { (self.funcs.destroy)(self.me) }
+        unsafe { self.funcs.IRMetalLibBinaryDestroy(self.me) }
     }
 }
 
-impl<'lib> IRMetalLibBinary<'lib> {
-    pub fn new(
-        lib: &'lib libloading::Library,
-    ) -> Result<IRMetalLibBinary<'lib>, Box<dyn std::error::Error>> {
+impl IRMetalLibBinary {
+    pub fn new(compiler: &IRCompiler) -> Result<IRMetalLibBinary, Box<dyn std::error::Error>> {
         unsafe {
-            let funcs = IRMetalLibBinaryFn {
-                create: lib.get(b"IRMetalLibBinaryCreate")?,
-                destroy: lib.get(b"IRMetalLibBinaryDestroy")?,
-                get_bytecode: lib.get(b"IRMetalLibGetBytecode")?,
-                get_bytecode_size: lib.get(b"IRMetalLibGetBytecodeSize")?,
-                synthesize_stage_in_function: lib.get(b"IRMetalLibSynthesizeStageInFunction")?,
-            };
-
-            let me = (funcs.create)();
-            Ok(Self { funcs, me })
+            let me = compiler.funcs.IRMetalLibBinaryCreate();
+            Ok(Self {
+                funcs: compiler.funcs.clone(),
+                me,
+            })
         }
     }
 
     pub fn get_byte_code(&self) -> Vec<u8> {
-        let size_in_bytes = unsafe { (self.funcs.get_bytecode_size)(self.me) };
+        let size_in_bytes = unsafe { self.funcs.IRMetalLibGetBytecodeSize(self.me) };
         let mut bytes = vec![0u8; size_in_bytes];
-        let written = unsafe { (self.funcs.get_bytecode)(self.me, bytes.as_mut_ptr()) };
+        let _written = unsafe {
+            self.funcs
+                .IRMetalLibGetBytecode(self.me, bytes.as_mut_ptr())
+        };
         bytes
```

Comment:
Assert that `_written` is equal to `size_in_bytes`, and/or resize/trim the `Vec` to `_written` before returning it.

Reply:
Postponed, though the modification is necessary to make the CI ("rightfully") happy.